Sources/SwiftBasicFormat/BasicFormat.swift

```diff
@@ -171,7 +171,29 @@ open class BasicFormat: SyntaxRewriter {
       }
     }
 
-    return false
+    let syntaxToken: Syntax = Syntax(token)
+    switch syntaxToken.keyPathInParent {
+    case \StringLiteralExprSyntax.closeQuote:
+      return token.tokenKind == .multilineStringQuote
+    default:
+      return false
+    }
+  }
+
+  /// Whether a trailing newline on `token` should be added.
+  open func requiresTrailingNewline(_ token: TokenSyntax) -> Bool {
+    // We don't want to add newlines inside string interpolation
+    if isInsideStringInterpolation(token) && token.tokenKind != .multilineStringQuote {
+      return false
+    }
+
+    let syntaxToken: Syntax = Syntax(token)
+    switch syntaxToken.keyPathInParent {
+    case \StringLiteralExprSyntax.openQuote:
+      return token.tokenKind == .multilineStringQuote
+    default:
+      return false
+    }
```

Comment:
I think you could just define `requiresTrailingNewline` in terms of `requiresNewline(between:and:)`: `requiresTrailingNewline(token)` is equivalent to to `requiresNewline(between: token, and: nextToken)`

Same for `requiresLeadingNewline`. That way we have a single source of truth for defining whether two tokens should be separated by newlines.